### ISSUE TITLE
etcdserver: request is too large - Large workflow support (like 2000+ node status)

### ISSUE THREAD

**Issue by guanhuichen** [kubeflow/pipelines#4175]
/kind feature

### What steps did you take:
kfp.dsl.graph_component recursion (might result in 2000+ node status) or kfp.dsl.ParallelFor with 2000 size input array

### What happened:
workflow stopped with error etcdserver: request is too large

### What did you expect to happen:
the run api returns the node status without error

### Environment:

How did you deploy Kubeflow Pipelines (KFP)?

KFP version: Build commit: 743746b

kfp version: 0.5.1
kfp-server-api: 0.5.0

### Anything else you would like to add:
https://github.com/argoproj/argo/blob/master/docs/offloading-large-workflows.md
argo v2.4 and later supports the secondary persistence (postgres). Will the offload feature be supported in the next future release?

**Comment by Ark-kun** [comment]
KFP is already using Argo v2.7.5
Do we need to enable the feature somehow?
I'm afraid Argo only compresses the status object. If the spec is too big this does not help.

**Comment by guanhuichen** [comment]
@Ark-kun Argo compressses the status object if it doesn't execced the 1M limit by etcdserver. Otherwise it will store the whole status object in the postgresql as the secondary store.  I tried to enable the persistence feature at argo workflowcontroller, the status returned form the run api (kfp) misses the "OffloadNodeStatusVersion" data and wont get the whole node status from the db

![image](https://user-images.githubusercontent.com/2447119/86990394-143aa480-c1cf-11ea-949a-88e91bc7046f.png)

**Comment by Ark-kun** [comment]
> Argo compressses the status object if it doesn't execced the 1M limit by etcdserver.

The problem is that with 2000 steps your `spec` object will also be very big.

**Comment by guanhuichen** [comment]
Oh i see, let me rephrase my words. In my case the spec could be small like ParallelFor with 2000 size input array or 2000 loop in recursion. 2000+ steps are meaning in the final running detail not in spec. Agreed that 2000 steps in spec object will also be very big. Users usually wont do that.

**Comment by xiehaoina** [comment]
I think it is a common case when using "ParallelFor" operators. Can this feature be supported in future releases?

**Comment by cp2587** [comment]
Hello,

Is there any update on this issue ? How did you circumvent the issue @guanhuichen ?
